### PR TITLE
Benchmarks prose

### PR DESCRIPTION
**Pull request by molpopgen**
Rough draft of text re: benchmark fig.

### CODE REVIEW

File: treestats_paper.tex [hunk 1]
```diff
@@ -1266,13 +1267,33 @@ \subsection*{Algorithms}
...
+\krt{need to sort out the big-O issue here}
+We used coalescent simulations to compare the performance of calculating Tajima's \citeyearpar{Tajima1989-de} $D$
+statistic from tree sequences to calculating it from integer matrices containing genotypes at all variable positions.
+\autoref{fig:stats_performance} shows that the tree sequence methods implemented in \texttt{tskit} process variant data
+three orders of magnitude faster than matrix-based approaches once the sample size is above $n \approx 500$ haploids.
+The expected number of mutations for each replicate is $10^4 \times \sum_{i=1}^{n-1}\frac{1}{i}$, and thus ranges from
+$\approx 32,000$ to $\approx 138,000$ \citep{Watterson1975-ej}.
+\autoref{fig:stats_performance} only considers the time required to calculate the statistic. It does not include the time
+required to build the genotype matrix, nor the memory required to store it.  For the largest samples size of $n=10^6$,
+the matrix is approximately 200 gigabytes, and thus not practical on many systems.
```
Comment (petrelharp) [inline treestats_paper.tex]: Perhaps note how much memory the tree sequence uses?

File: treestats_paper.tex [hunk 2]
```diff
@@ -1266,13 +1267,33 @@ \subsection*{Algorithms}
...
     \includegraphics{tskit_stat_benchmarks/benchmarks_without_copy_longer_genome.pdf}
     \caption{Comparison of time required to compute site statistics
-        between matrix and tree based methods. [Details]
+        between matrix and tree based methods. For each sample size, a single replicate
+        was obtained using \texttt{msprime} with scaled neutral mutation rate $\theta = 4N_e\mu$ and
+        scaled recombination rate $\rho = 4Nr$ both equal to $10^4$. For each replicate, Tajima's $D$
+        \citep{Tajima1989-de} statistic was calculated using \texttt{tskit}, \texttt{libsequence}
+        \citep{Thornton2003-wj}, and \texttt{scikit-allel} \krt{need cite for this}.
+        The solid green line shows the result of fitting the timing data to a model based on the expected complexity of the 
+        algorithm used by \texttt{tskit}. \krt{need to revisit the correctness of this}
```
Comment (petrelharp) [inline treestats_paper.tex]: Should be N_e not N in the defn of rho? And, should say that Ne = 1e4?

**Comment by petrelharp** [comment]
I've updated the O( ) stuff. It wasn't right because we had a model for total time to compute a branch stat, while what we were fitting was per-site time to fit a site stat. I adjusted this, and also had it do the fitting on a log-log scale (since previously most of the fit was being determined by the biggest value). Note that the model is the same because in the sims mu = r.  The fit is somewhat better, I think.

File: treestats_paper.tex (hunk 2 above)
Comment (molpopgen) [reply treestats_paper.tex]: Will fix that.  I'll leave out the actual value of Ne, as it is a continuous-time coalescent sim, and so it is just a fudge factor anyways.

File: treestats_paper.tex (hunk 1 above)
Comment (molpopgen) [reply treestats_paper.tex]: This is not easily knowable.  The tree sequence objects don't have a member function to return their allocated size.  The size of the genotype matrix can be obtained directly from the sample size + number of mutations.  

One option would be to rerun the benchmarks, and output the size of node + edge tables.  Then, we know that there are 5 or 6 arrays making up the linked list for a tree, and each is `len(node table)`.  From that, I can get the size pretty much right.  Thoughts?

File: treestats_paper.tex (hunk 2 above)
Comment (petrelharp) [reply treestats_paper.tex]: True, but I vote to actually put in the value of Ne anyhow, to make things more concrete for people who don't live in coalescent land.

Comment (molpopgen) [reply treestats_paper.tex]: Done.

File: treestats_paper.tex (hunk 1 above)
Comment (petrelharp) [reply treestats_paper.tex]: Well, we don't need to be totally precise here. Maybe it's enough to either (a) open python, and the tree sequence in it, and see how much memory is used, or (b) just say "the tree sequence uses much less memory"?

Comment (molpopgen) [reply treestats_paper.tex]: The exact tree sequences are not stored, but I could fudge it by simulating one real quick...

Comment (molpopgen) [reply treestats_paper.tex]: Done

**Comment by petrelharp** [comment]
feel free to merge when you want, btw

**Pull request merged**